task/backend/analytical_storage.go

```diff
@@ -67,25 +67,20 @@ func (as *AnalyticalStorage) FinishRun(ctx context.Context, taskID, runID influx
 
 		// log an error if we have incomplete data on finish
 		if !run.ID.Valid() ||
-			run.ScheduledFor == "" ||
-			run.StartedAt == "" ||
-			run.FinishedAt == "" ||
 			run.Status == "" {
 			as.logger.Error("Run missing critical fields", zap.String("run", fmt.Sprintf("%+v", run)), zap.String("runID", run.ID.String()))
 		}
 
 		fields := map[string]interface{}{}
 		fields[runIDField] = run.ID.String()
-		fields[startedAtField] = run.StartedAt
-		fields[finishedAtField] = run.FinishedAt
-		fields[scheduledForField] = run.ScheduledFor
-		if run.RequestedAt != "" {
-			fields[requestedAtField] = run.RequestedAt
-		}
-
-		startedAt, err := run.StartedAtTime()
-		if err != nil {
-			startedAt = time.Now()
+		// cast to UTC here and when we read from the database so that run timestamps are always in UTC format
+		fields[startedAtField] = run.StartedAt.UTC().Format(time.RFC3339Nano)
+		fields[finishedAtField] = run.FinishedAt.UTC().Format(time.RFC3339Nano)
+		fields[scheduledForField] = run.ScheduledFor.UTC().Format(time.RFC3339)
+
+		startedAt := run.StartedAt
+		if !run.IsStarted() {
+			startedAt = time.Now().UTC()
```

Comment:
This code previously checked for both an empty StartedAt and RequestedAt. I removed the check for the empty RequestedAt, since it will just be written to the object as the zero time if it does not exist, but I left in the check for StartedAt since that will be used as the timestamp for the call to `NewPoint` below. I also added utility functions to check if a run has been started or finished, but not for requested or scheduled. Open to suggestions on this, we could alternatively use a single IsZeroTime function, remove these functions, etc.

Reply:
that is probably fine, The run should not get to this point without being started since the we are in the `FinishRun` method.